### ISSUE TITLE
Suggest requestHandler config change in migration docs

### ISSUE THREAD

**Issue by richdougherty** [playframework/playframework#4477]
If you're not using `GlobalSettings` it's faster to configure

```
 play.http.requestHandler = "play.api.http.DefaultHttpRequestHandler"
```

as described in the `Scala/JavaHttpRequestHandler.md` docs. We should mention this in the migration docs so that people know to change the setting.

**Comment by gmethvin** [comment]
Might also be worth mentioning this is only for runtime DI. If you use compile time DI with `BuiltInComponents` then `DefaultHttpRequestHandler` is the default.

**Comment by wsargent** [comment]
Looks like @dotta did master with 91e5c7075e6abc3ab06b232eb9fc2a655e815368 as part of https://github.com/playframework/playframework/pull/4308

**Comment by richdougherty** [comment]
That looks like a change to the `ScalaRequestHandlers.md` file. What I mean is that we should add a mention to `Migration24.md` that tells people to change the setting:

> The `HttpRequestHandler` that Play uses by default delegates to the legacy `GlobalSettings` methods. If you're not using `GlobalSettings` in your application then you can increase performance slightly by changing the handler. You can do that by adding the following to your settings:
>
>    play.http.requestHandler = "play.http.DefaultHttpRequestHandler"
>
> Only do this if you're _not_ using a `GlobalSettings` object. This change will stop some of the methods in your `GlobalSettings` object from being called.

**Comment by wsargent** [comment]
Merged #4527
